atomic_reactor/plugins/pre_pyrpkg_fetch_artefacts.py

```diff
@@ -26,13 +33,80 @@ def __init__(self, workflow):
         constructor
 
         :param workflow: DockerBuildWorkflow instance
-        :param command: str, command to use to get artefacts (e.g. 'make sources')
-                             it is executed in cloned git repo
+        :type workflow: atomic_reactor.inner.DockerBuildWorkflow
         """
         # call parent constructor
         super(DistgitFetchArtefactsPlugin, self).__init__(workflow)
         self.command = self.workflow.conf.sources_command
 
+    def _collect_exploded_sources_files(self, build_dir_path: Path) -> List[Path]:
+        """Collect the fetched exploded sources.
+
+        :param build_dir_path: from which directory to find the exploded sources file.
+        :type build_dir_path: pathlib.Path
+        :return: a list of file names got from the source-repos file. An empty
+            list will be returned if no source-repos exist, that means the
+            container repo does not have any exploded sources.
+        :rtype: list[pathlib.Path]
+        :raises ValueError: if any line of source-repos cannot be parsed or the
+            file cannot be found from the build_dir. Note that, generally,
+            neither of these issues should be captured by this method, since
+            it runs after rhpkg generates the tarballs from the repos listed
+            within source-repos and any potential issue should be handled by
+            rhpkg.
+        """
+        source_repos = build_dir_path / EXPLODED_SOURCES_FILE
+        filenames = []
+        if not source_repos.exists():
+            return filenames
+        regex = re.compile(r'^([\S]+)\s+([0-9a-f]+)$')
```

Comment:
If I'm looking at the right rhpkg version, the regex they use is `r'^([\S]+)\s+([\S]+)$'` (which could be simplified to `r'^(\S+)\s+(\S+)$'`). It seems that our regex puts additional restrictions on the git ref, is that OK?

And another detail is that rhpkg matches this regex against `line.strip()`, so it allows extra whitespace on both ends. Now that I think about it, the rhpkg approach is just an unnecessarily complicated way to do `line.split()` and check that it splits to 2 elements :smile: